laravel/database/migrations/2017_09_26_141754_create_statistics_table.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateStatisticsTable extends Migration
```

Comment:
This is an example of database `migrations`. This is how you create database tables with versioning. The boilerplate for these migrations should be created from the commandline

For example:

```
vagrant ssh
cd /vagrant/laravel
php artisan make:migration create_statistics_table
```

The class defineds two methods, up and down. The `up()` method is called when upgrading the database to a newer schema, `down()` is called when downgrading- so the logic in the down method should revert any changes made in the up method.

Laravel stores the details of which migrations have actually been executed inside the database in a special table called `migrations`. If the table doesn't exist it assumes the database is new and will run all migrations- if the table does exist, it will only execute the migrations it has not already run.

To upgrade

```
vagrant ssh
cd /vagrant/laravel
php artisan migrate
```

To downgrade

```
vagrant ssh
cd /vagrant/laravel
php artisan db:rollback
```

To blow everything away and start again

```
vagrant ssh
cd /vagrant/laravel
php artisan db:fresh
```